app/javascript/components/schedule-form/helper.js

```diff
@@ -0,0 +1,197 @@
+  resources.map((item) => ({ label: item.description, value: item.id })));
+
+/** Function to load the object items when the 'Object Type' drop-down-list selection is changed  */
+export const objectTypeChange = (value, setData, data) => {
+  if (value && data.hideAutomationFields === false) {
+    data.hideObjectItem = false;
+    http.post(`/ops/fetch_target_ids/?target_class=${value}`).then(({ targets }) => {
+      setData({
+        ...data,
+        objectItemOptions: targets.map((item) => ({ label: item[0], value: item[1] })),
+      });
+    });
+  }
+};
+
+/** Function to load the target options when the 'Filter' drop-down-list selection is changed  */
+export const subActionChange = (value, setData, data) => {
+  data.initialValues.filter_type = value;
+  data.hideTarget = true;
+  if (value !== 'all') {
+    data.hideTarget = false;
+    http.post('/ops/schedule_form_filter_type_field_changed/new',
+      { filter_type: value, action_type: data.initialValues.action_typ }).then((response) => {
+      const targets = response.filtered_item_list;
+      const targetOptions2 = isArray(targets[0])
+        ? targets.map((item) => ({ label: item[1], value: item[0] }))
+        : targets.map((item) => ({ label: item, value: item }));
+      setData({
+        ...data,
+        targetOptions: targetOptions2,
+      });
+    });
+  } else {
+    setData({
+      ...data,
+    });
+  }
+};
+
+/** Run options data needed for the run drop-down-list */
+const runTypes = {
+  hours: { label: __('Hours'), type: 'Hourly', options: [2, 3, 4, 6, 8, 12] },
+  daily: { label: __('Days'), type: 'Daily', options: [2, 3, 4, 5, 6] },
+  weekly: { label: __('Weeks'), type: 'Weekly', options: [2, 3, 4] },
+  monthly: { label: __('Months'), type: 'Monthly', options: [2, 3, 4, 5, 6] },
+};
+
+/** Function to load the `Run` Select field options */
+export const runOptions = [{
+  label: __('Once'),
+  value: 'Once',
+},
+{
+  label: runTypes.hours.label,
+  value: runTypes.hours.type,
+},
+{
+  label: runTypes.daily.label,
+  value: runTypes.daily.type,
+},
+{
+  label: runTypes.weekly.label,
+  value: runTypes.weekly.type,
+},
+{
+  label: runTypes.monthly.label,
+  value: runTypes.monthly.type,
+}];
+
+/** Function to return the data needed for the selected run drop-down-list */
+const timerOptions = ({ options, label }) => {
+  const data = options.map((value) => ({
+    value,
+    label: `${value} ${label}`,
+  }));
+  return data;
+};
+
+/** Function to handle the run drop-down-list on change event */
+export const runChange = (value) => {
+  switch (value) {
+    case runTypes.hours.type:
+      return timerOptions(runTypes.hours);
+    case runTypes.daily.type:
+      return timerOptions(runTypes.daily);
+    case runTypes.weekly.type:
+      return timerOptions(runTypes.weekly);
+    case runTypes.monthly.type:
+      return timerOptions(runTypes.monthly);
+    default:
+      return [];
+  }
```

Comment:
hey, I just thought of a change. can you try this..

`timerOptions(runTypes[value].type)`